content/page/pods.md

```diff
@@ -13,30 +13,58 @@ To launch a pod using the container [image](https://quay.io/repository/openshift
 `quay.io/openshiftlabs/simpleservice:0.5.0` and exposing a HTTP API on port `9876`, execute:
 
 ```bash
-$ kubectl run sise --image=quay.io/openshiftlabs/simpleservice:0.5.0 --port=9876
+kubectl run sise --image=quay.io/openshiftlabs/simpleservice:0.5.0 --port=9876
 ```
+```cat
+pod/sise created
+```
+
+***Note: Deprecation Warning!***
+Older releases of `kubectl` will produce a [deployment](/deployments/) resource as the result of the provided `kubectl run` example, while newer releases produce a single `pod` resource.  The example commands in this section should still work (assuming you substitute your own pod name) - but you'll need to run `kubectl delete deployment sise` at the end of this section to clean up.
 
-We can now see that the pod is running:
+Check to see if the pod is running:
 
 ```bash
-$ kubectl get pods
-NAME                      READY     STATUS    RESTARTS   AGE
-sise-3210265840-k705b     1/1       Running   0          1m
+kubectl get pods
+```
+```cat
+NAME    READY     STATUS    RESTARTS   AGE
+sise    1/1       Running   0          1m
+```
 
-$ kubectl describe pod sise-3210265840-k705b | grep IP:
-IP:                     172.17.0.3
+If the above output returns a longer pod name, make sure to use it in the following examples (in place of `sise`)
+
+This container image happens to include a copy of `curl`, which provides an additional way to verify that the primary webservice process is responding (over the local net at least):
+
+```bash
+kubectl exec sise -t -- curl -s localhost:9876/info
+```
+```cat
+{"host": "localhost:9876", "version": "0.5.0", "from": "127.0.0.1"}
 ```
 
-From within the cluster (e.g. via [`oc rsh`](https://docs.openshift.com/container-platform/latest/cli_reference/openshift_cli/developer-cli-commands.html#rsh)) this pod is accessible via the pod IP `172.17.0.3`,
-which we've learned from the `kubectl describe` command above:
+From within the cluster (e.g. via `kubectl exec` or [`oc rsh`](https://docs.openshift.com/container-platform/latest/cli_reference/openshift_cli/developer-cli-commands.html#rsh)) this pod will also be directly accessible via it's associated pod IP `172.17.0.3`
+
+```bash
+kubectl describe pod sise | grep IP:
+```
+```cat
+IP:                     172.17.0.3
+```
 
+The kubernetes API provides an additional opportunity to connect directly to pods using `curl`:
 ```bash
-[cluster] $ curl 172.17.0.3:9876/info
-{"host": "172.17.0.3:9876", "version": "0.5.0", "from": "172.17.0.1"}
+export OPENSHIFT_API="https://$(kubectl config get-clusters | tail -n 1)"
```

Comment:
It bugs me that you are using OPENSHIFT_API and $(oc whoami -t) here. It is not given that a user will have OpenShift/oc at this point.

Reply:
Maybe another way to do this would be to use `kubectl proxy` ?

```bash
kubectl proxy --port=8080
```
```cat
Starting to serve on 127.0.0.1:8080
```
In another terminal window
```bash
curl localhost:8080/api/v1/namespaces/default/pods/sise/proxy/info
```
```cat
{"host": "localhost:8080", "version": "0.5.0", "from": "10.244.0.1"}
```

Reply:
Good feedback. I'd prefer to use kubectl instead of oc in this example if possible.  Have not solved that yet.  I'll make a note and file an issue to track it.

Using the proxy is a great suggestion - we should definitely mention it.  I was hoping to keep the interaction all within a single terminal (if possible) in order to simplify test automation efforts for this content.

Reply:
filed https://github.com/openshift-evangelists/kbe/issues/63

Reply:
Removed `oc whoami -t` from the example: https://github.com/openshift-evangelists/kbe/commit/9876def8d1c58185102d349c2ead30e3754a3a88